pennylane/tape/qscript.py

```diff
@@ -1331,6 +1334,21 @@ def from_queue(cls, queue):
         return cls(*process_queue(queue))
 
 
+class SpecsDict(dict):
+    """A dictionary to track and warn about deprecated keys"""
+
+    deprecated_keys = ("gate_types", "num_operations", "num_used_wires", "depth")
+    new_keys = ("gate_types", "num_gates", "num_wires", "depth")
```

Comment:
Could also make this a `dict` between deprecated keys and new keys, making the lookup simpler.
However, as this is a warning/exceptional evaluation path, I suppose it does not matter too much.

Reply:
Yes it would make the lookup simple. Initially the message was only displaying the deprecated keys, I added the new keys after the fact and completely missed this potential optimization.

Reply:
It really works both ways, no worries to optimize...

Reply:
Added 👍🏼